packages/wallet/test/e2e/SingleAddressWallet/cip30mapping.test.ts

```diff
@@ -0,0 +1,138 @@
+
+  beforeAll(async () => {
+    // CREATE A WALLET
+    const keyAgent = await keyAgentReady;
+    wallet = new SingleAddressWallet(
+      { name: 'Test Wallet' },
+      {
+        assetProvider,
+        keyAgent,
+        stakePoolSearchProvider,
+        timeSettingsProvider,
+        txSubmitProvider,
+        walletProvider
+      }
+    );
+
+    mappedWallet = createCip30WalletApiFromWallet(wallet, {
+      keyAgent,
+      logger: console
+    });
+  });
+
+  afterAll(() => {
+    wallet.shutdown();
+  });
+
+  it('should correctly pull wallet balance correctly from the provider before starting the test', async () => {
+    await firstValueFrom(wallet.balance.total$);
+    expect(wallet.balance.total$.value?.coins).toBeGreaterThanOrEqual(0n);
+  });
+
+  describe('API Methods', () => {
+    test('api.getNetworkId', async () => {
+      const cip30NetworkId = await mappedWallet.getNetworkId();
+      expect(cip30NetworkId).toEqual(Number(process.env.NETWORK_ID));
+    });
+
+    test('api.getUtxos', async () => {
+      const utxos = await mappedWallet.getUtxos();
+      expect(coreToCsl.utxo(utxos!)).not.toThrow();
+    });
+
+    test('api.getBalance', async () => {
+      const balanceCborBytes = Buffer.from(await mappedWallet.getBalance(), 'hex');
+      expect(() => CSL.Value.from_bytes(balanceCborBytes)).not.toThrow();
+    });
+
+    test('api.getUsedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUsedAddresses();
+      const [{ address: walletUsedAddresses }] = await firstValueFrom(wallet.addresses$);
+      expect(cipUsedAddressess).toEqual(walletUsedAddresses);
+    });
+
+    test('api.getUnusedAddresses', async () => {
+      const cipUsedAddressess = await mappedWallet.getUnusedAddresses();
+      expect(cipUsedAddressess).toEqual([]);
+    });
+
+    test('api.getChangeAddress', async () => {
+      const cipChangeAddress = await mappedWallet.getChangeAddress();
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address.toString());
+      if (!parsedAddress) {
+        throw new Error('No wallet address');
+      }
+
+      expect(cipChangeAddress).toEqual(Buffer.from(parsedAddress.to_bytes()).toString('hex'));
+    });
+
+    test('api.getRewardAddresses', async () => {
+      const cipRewardAddresses = await mappedWallet.getRewardAddresses();
+      const [{ rewardAccount }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(rewardAccount.toString());
+
+      expect(cipRewardAddresses).toEqual([Buffer.from(parsedAddress!.to_bytes()).toString('hex')]);
+    });
+
+    test('api.signTx', async () => {
+      const txInternals = await createTxInternals(wallet);
+      const finalizedTx = await wallet.finalizeTx(txInternals);
+
+      const cslTx = coreToCsl.tx(finalizedTx).to_bytes();
+      expect(await mappedWallet.signTx(Buffer.from(cslTx).toString('hex'))).not.toThrow();
```

Comment:
I think this would be a better test
```suggestion
      const signatures = Buffer.from(await mappedWallet.signTx(Buffer.from(cslTx).toString('hex')), 'hex')
      expect(() => CSLTransactionWitnessSet.from_bytes(signatures)).not.toThrow();
```